src/main/java/de/tum/in/www1/artemis/repository/ComplaintRepository.java

```diff
@@ -234,53 +249,67 @@ SELECT count(c) FROM Complaint c
-                 and :#{#groupName} member of complaint.result.assessor.groups
-                 and complaint.result.participation.exercise.id = :exerciseId
-                 and complaint.result.completionDate IS NOT NULL
-                 and complaint.accepted IS NOT NULL
-             GROUP BY complaint.complaintResponse.reviewer.id
+                 c.complaintType = 'COMPLAINT'
+                 and :#{#groupName} member of a.groups
+                 and e.id = :exerciseId
+                 and r.completionDate IS NOT NULL
+                 and c.accepted IS NOT NULL
+             GROUP BY cr.reviewer.id
              """)
     List<TutorLeaderboardComplaintResponses> findTutorLeaderboardComplaintResponsesByExerciseId(@Param("groupName") String groupName, @Param("exerciseId") long exerciseId);
 
+    @Query("""
+            SELECT
+             new de.tum.in.www1.artemis.domain.leaderboard.tutor.TutorLeaderboardComplaintResponses(
+                 cr.reviewer.id,
+                 count(c),
+                 sum(e.maxPoints)
+             )
+             FROM
+                Complaint c join c.complaintResponse cr join c.result r join r.participation p join p.exercise e join e.exerciseGroup eg join eg.exam ex  join r.assessor a
+             WHERE
+                 c.complaintType = 'COMPLAINT'
+                 and :#{#groupName} member of a.groups
+                 and ex.id = :examId
+                 and r.completionDate IS NOT NULL
+                 and c.accepted IS NOT NULL
+             GROUP BY cr.reviewer.id
+             """)
+    List<TutorLeaderboardComplaintResponses> findTutorLeaderboardComplaintResponsesByExamId(@Param("groupName") String groupName, @Param("examId") long examId);
+
     /*
      * @Query(""" SELECT new de.tum.in.www1.artemis.domain.leaderboard.tutor.TutorLeaderboardMoreFeedbackRequests( null, result.assessor, sum(exercise.max_points), course.id ) FROM
-     * Complaint complaint left join complaint.result result WHERE (complaint.complaint_type = 'MORE_FEEDBACK') and result.participation.exercise.course.teachingAssistantGroupName
-     * member of result.assessor.groups and result.completionDate is not null and result.participation.exercise.course.id = :#{#courseId} and
-     * result.participation.exercise.discriminator in ('M', 'T', 'F', 'P') and (g.user_id = u.id) and (g.groups = c.teaching_assistant_group_name) and (c.id = e.course_id) and
-     * (p.exercise_id = e.id) and (e.discriminator in('M', 'T', 'F', 'P')) and (r.participation_id = p.id) and (r.assessor_id = u.id) and (cp.result_id = r.id) and
-     * (r.completion_date IS NOT NULL) GROUP BY result.assessor.id, course.id """) List<TutorLeaderboardMoreFeedbackRequests>
-     * findTutorLeaderboardMoreFeedbackRequestsByCourseId(@Param("courseId") long courseId);;
+     * Complaint complaint left join c.result result WHERE (c.complaint_type = 'MORE_FEEDBACK') and result.participation.exercise.course.teachingAssistantGroupName member of
```

Comment:
What is that commented out query?

Reply:
thats the original view query from the database. I removed them.